share/completions/p4.fish

```diff
@@ -0,0 +1,726 @@
+
+function __fish_print_p4_streams
+    # I do not have an example of p4 streams output
+    #p4 streams 2> /dev/null
+end
+
+#########################################################
+
+function __fish_print_p4_users
+    string replace -ar '(^\S+) <([^>]+)> \(([^\)]+)\).*$' '$1'\t'$3 <$2>' (p4 users)
+end
+
+#########################################################
+### completion helpers
+#########################################################
+
+function __fish_print_p4_commands_list
+    set -l commands add annotate attribute branch branches change changes changelist changelists clean client clients copy counter counters cstat delete depot depots describe diff diff2 dirs edit filelog files fix fixes flush fstat grep group groups have help info integ integrate integrated interchanges istat job jobs key keys label labels labelsync list lock logger login logout merge move opened passwd populate print protect protects prune rec reconcile rename reopen resolve resolved revert review reviews set shelve status sizes stream streams submit sync tag tickets unlock unshelve update user users where workspace workspaces
+    for i in $commands
+        echo $i
+    end
+end
+
+#########################################################
+
+function __fish_print_p4_env_vars
+    set -l env_vars P4CLIENT P4PORT P4PASSWD P4USER P4CONFIG P4DIFF P4EDITOR P4MERGE P4CHARSET P4TRUST P4PAGER PWD TMP TEMP P4TICKETS P4LANGUAGE P4LOGINSSO P4COMMANDCHARSET P4DIFFUNICODE P4MERGEUNICODE P4CLIENTPATH P4AUDIT P4JOURNAL P4LOG P4PORT P4ROOT P4DEBUG P4NAME P4SSLDIR
+    for var in $env_vars
+        echo $var
+    end
+end
+
+#########################################################
+
+function __fish_print_p4_workspace_changelists
+    set -l client (__fish_print_p4_client_name)
+    if test -n "$client"
+        __fish_print_p4_changelists -c $client $argv
+    end
+end
+
+#########################################################
+
+function __fish_print_p4_pending_changelists
+    __fish_print_p4_workspace_changelists -s pending
+end
+
+#########################################################
+
+function __fish_print_p4_shelved_changelists
+    __fish_print_p4_workspace_changelists -s shelved
+end
+
+#########################################################
+
+function __fish_print_p4_file_types
+    set -l base_types text binary symlink apple resource unicode utf16
+    set -l modifiers +m +w +x +k +ko +l +C +D +F +S +X
+    set -l mod_desc 'always set modtime' 'always writeable' 'exec bit set' '$Keyword$ expansion of Id, Header, Author, Date, DateUTC, DateTime, DateTimeUTC, DateTimeTZ, Change, File, Revision' '$Keyword$ expansion of Id, Header only' 'exclusive open: disallow multiple opens' 'server stores compressed file per revision' 'server stores deltas in RCS format' 'server stores full file per revision' 'server stores only single head rev., or specify number to <n> of revisions' 'server runs archive trigger to access files'
+    for i in $base_types
```

Comment:
This might be nicer as a multiline `printf`, e.g. 

```fish
for i in base_types
    printf '%s\t%s\n' $i+m 'always set modtime' \
                            $i+w 'always writeable' \
# [...]
```

Which would also eliminate the `math` call (which forks off bc).